### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by MarcRoopchand**
Split up files so that they can be generated for Java

### CODE REVIEW

**Comment by kgrgreer** [comment]
You've added back the class list in genjava which was previously removed.

File: tools/genjava.js [hunk 1]
```diff
@@ -15,123 +15,123 @@
  * limitations under the License.
  */
 
-// enable FOAM java support.
-global.FOAM_FLAGS = { 'java': true, 'debug': true };
+ // enable FOAM java support.
```
Comment (kgrgreer) [inline tools/genjava.js]: Why did you add a blank space to the beginning of each line?

Comment (MarcRoopchand) [reply tools/genjava.js]: My editor was being very weird. I just reverted to the last commit. Not sure where that white space came from
